### PR TITLE
Version bump action

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR creates a GH action to bump versions plugin main file, `info.json` and `readme.txt`

### CODE REVIEW

File: packages/version-bump/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+	"name": "@eventespresso-actions/version-bump",
+	"version": "0.0.1",
+	"private": true,
+	"main": "lib/index.js",
+	"scripts": {
+		"build": "tsc",
+		"format": "prettier --write **/*.ts",
+		"format-check": "prettier --check **/*.ts",
+		"lint": "eslint src/**/*.ts",
```
Comment (nicolad) [inline packages/version-bump/package.json:10]: :+1:

File: packages/version-bump/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	const releaseTypes = releaseTypesStr ? JSON.parse(releaseTypesStr) : DEFAULT_RELEASE_TYPES;
+
+	return {
+		infoJsonFile,
+		mainFile,
+		readmeFile,
+		releaseTypes,
+		type,
+	};
+}
```
Comment (nicolad) [inline packages/version-bump/src/utils.ts:55]: nice extraction over the old approach

File: packages/version-bump/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ * MINOR    ([0-9]+)                 MUST match & capture a number
+ * DOT      \.                       MUST match a period
+ * PATCH    ([0-9]+)                 MUST match & capture a number
+ * DOT      \.?                      maybe match a period
+ * RELEASE  (alpha|beta|rc|p|decaf)* maybe match one of values in brackets
+ * DOT      \.?                      maybe match a period
+ * BUILD    ([0-9]*)                 maybe match & capture a number
+ * @see: https://regex101.com/r/5nSgf3/1/
+ */
+export const EE_VERSION_REGEX = /^(?<major>[0-9]+)\.(?<minor>[0-9]+)\.(?<patch>[0-9]+)\.?(?<release>alpha|beta|rc|p|decaf)*\.?(?<build>[0-9]*)$/;
```
Comment (nicolad) [inline packages/version-bump/src/utils.ts:79]: maybe to cover these regex with tests?
https://jestjs.io/docs/en/expect#tomatchregexp--string

**Review by nicolad — APPROVED** [review]
Amazing improvement in readability and comprehension over the old PHP script

File: packages/version-bump/src/utils.ts (hunk 3 above)
Comment (manzoorwanijk) [reply packages/version-bump/src/utils.ts:79]: Yeah, I agree

File: packages/version-bump/src/tests/main.test.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		outputVer: '4.10.9.beta',
+	},
+	{
+		type: 'minor',
+		inputVer: '4.10.9.rc.011',
+		outputVer: '4.11.0.p',
+	},
+];
+
+describe('version bump main tests', () => {
```
Comment (nicolad) [inline packages/version-bump/src/tests/main.test.ts:27]: NOICEE

**Review by nicolad — APPROVED** [review]

File: packages/version-bump/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+              with:
+                  type: minor
+                  main-file: src/espresso.php
+                  readme-file: src/readme.txt
+                  info-json-file: info.json
+              # then you can use the `new-version` output in subequent steps/jobs
+              - name: Some Step
+                uses: some@action
+                with:
+                  version: ${{ steps.bump-version.outputs.new-version }}
```
Comment (tn3rb) [inline packages/version-bump/README.md:50]: very cool

File: packages/version-bump/src/tests/main.test.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+		type: 'pre_release',
+		inputVer: '4.10.9.rc.011',
+		outputVer: '4.10.9.beta',
+	},
+	{
+		type: 'minor',
+		inputVer: '4.10.9.rc.011',
+		outputVer: '4.11.0.p',
+	},
+];
```
Comment (manzoorwanijk) [inline packages/version-bump/src/tests/main.test.ts:25]: These are just example cases. It needs plenty of test cases which can be added in separate PRs.

File: packages/version-bump/src/tests/utils.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+export const getMockedFileContents = (path: string, version?: string): string => {
+	switch (path) {
+		case 'main-file.php':
+			return `/*
+            Plugin Name:Event Espresso
+            Plugin URI: some-uri
+            Version: ${version || '4.10.9.rc.011'}
+            Author: Event Espresso
+            Text Domain: event_espresso
+            */`;
```
Comment (tn3rb) [inline packages/version-bump/src/tests/utils.ts:10]: noice

File: packages/version-bump/src/utils.ts (hunk 2 above)
Comment (tn3rb) [reply packages/version-bump/src/utils.ts:55]: i concur

**Review by tn3rb — APPROVED** [review]
### super amazing job!!! 💯

**Pull request merged**